vhost-device-scmi/src/devices/common.rs

```diff
@@ -453,6 +482,68 @@ pub trait SensorT: Send {
             _ => Result::Err(ScmiDeviceError::UnsupportedRequest),
         }
     }
+
+    /// Returns the notification messages from device.
+    ///
+    /// Usually need to redefine this. Different sensor may has different way to
+    /// get notification.
+    fn reading_device(&self, _device_index: u32) -> DeviceResult {
+        Ok(vec![])
+    }
+
+    /// Enable Sensor notify function.
+    ///
+    /// Usually need to redefine this. Different sensor will do different
+    /// configuration set to enable notify funciton.
+    fn enable_notify(&self) -> Result<(), DeviceError> {
+        Ok(())
+    }
+
+    /// Disable Sensor notify function.
+    ///
+    /// Usually need to redefine this. Different sensor will do different
+    /// configuration set to disable notify funciton.
+    fn disable_notify(&self) -> Result<(), DeviceError> {
+        Ok(())
+    }
+
+    /// Get notify device fd
+    ///
+    /// This fd is used for getting notification.
+    /// It should be registered in epoll handler.
+    fn get_notify_fd(&self) -> Option<RawFd> {
+        None
+    }
+
+    /// Set a id for this device.
+    ///
+    /// Usually no need to redefine this.
+    /// Sensor id is increasing during registration.
+    fn set_id(&mut self, id: usize) {
+        self.sensor_mut().sensor_id = id;
+    }
+
+    /// Get id of this device.
+    ///
+    /// Usually no need to redefine this.
+    fn get_id(&self) -> usize {
+        self.sensor().sensor_id
+    }
+
+    /// Get a nofication message value from this sensor.
+    ///
+    /// The sensor's default implement only support SENSOR_UPDATE notification.
```

Comment:
Oh sorry, in currently, we only implement this notification.
And there still remains "SENSOR_TRIP_POINT_EVENT" notification which we haven't implemented.
We are not familiar with this command, maybe it can be implemented in future.

Reply:
well